snuba/web/query.py

```diff
@@ -43,6 +43,7 @@
 
 class RawQueryResult(NamedTuple):
     result: Result
+    applied_sampling_rate: Optional[float]
```

Comment:
`This decision is going to need to be made prior to the query splitting.`
I don't think this is possible today. Splitting happens before anything related with query processing. See my longer comment about why split needs a complete redesign to work with query processing. Split today does not even know about datasets, it is still the same as it was in the first version of Snuba and applied equally to all queries.
Splitting is broken with sampling already and it was already before we built datasets, think about the column splitter (queries a set of event_ids over a limited set of columns, then runs a second query over all columns specifying those event_id) if someone provides a sampling rate and we decide to split the query, the effective sampling will be the product of the two, since the sampling rate is applied to both queries and the conditions to the second query are the result of the first.

So I disagree on the point that the same sampling rate will have to be applied to all subqueries. It depends on the split logic, that's what makes rebuilding splitting so dependent on having a good end to end query processing system.

I really don't think it is useful to rebuild query splitting till we do not have a viable query processing system for individual queries.